src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,307 @@
+     * @return The relevant type of Command object.
+     * */
+    public static Command parseCommands(String command, TaskList tasks) throws TobiasException {
+        if (command.equals("hello")) {
+            return new ReplyCommand("Howdy there partner !");
+        } else if(command.equals("list")) {
+            return new ListCommand();
+        } else if (command.startsWith("todo")) {
+            if (command.equals("todo")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(4,5);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after todo!");
+            }
+
+            String description = command.substring(5);
+
+            return new TodoCommand(description);
+        } else if (command.startsWith("deadline")) {
+            if (command.equals("deadline")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(8,9);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after deadline!");
+            }
+
+            int byIndex = command.indexOf("/by");
+
+            if (byIndex == -1) {
+                throw new TobiasException("Hey, please use '/by' to specify the deadline!!");
+            }
+
+            String description = command.substring(9, byIndex);
+
+            int deadlineIndex = byIndex+4;
+
+            if (deadlineIndex > command.length()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            String deadline = command.substring(byIndex + 4);
+
+            LocalDateTime dd = dateFromString(deadline);
+
+            if (deadline.isEmpty() || deadline.isBlank()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            return new DeadlineCommand(description, dd);
+        } else if (command.startsWith("event")) {
+            if (command.equals("event")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(5,6);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after event!");
+            }
+
+            int fromIndex = command.indexOf("/from");
+            int toIndex = command.indexOf("/to");
+
+            if (fromIndex == -1 && toIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from and /to date/day/time!!");
+            }
+
+            if (fromIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from date/day/time!!");
+            }
+
+            if (toIndex == -1) {
```

Comment:
This method is too long. Isolate command format checking from this matching stage. Perhaps make a method for each of the commands with non-trivial parameters?